From be97dfe2ba398f9063faadf36189dcf368ce9873 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:49:34 -0400
Subject: [iommu] vt-d: Store PCI segment number in struct intel_iommu

Message-id: <20140905014934.31297.16938.stgit@gir.stowe>
Patchwork-id: 91331
O-Subject: [RHEL7.1 PATCH 38/65] iommu/vt-d: Store PCI segment number in struct intel_iommu
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 67ccac41fafda88492620f4c0a30d4ccb2eb7767

commit 67ccac41fafda88492620f4c0a30d4ccb2eb7767
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sun Mar 9 13:49:45 2014 -0700

    iommu/vt-d: Store PCI segment number in struct intel_iommu

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/dmar.c b/drivers/iommu/dmar.c
index ddfbd0e..c38a44f 100644
--- a/drivers/iommu/dmar.c
+++ b/drivers/iommu/dmar.c
@@ -953,6 +953,7 @@ static int alloc_iommu(struct dmar_drhd_unit *drhd)
  }
  iommu->agaw = agaw;
  iommu->msagaw = msagaw;
+ iommu->segment = drhd->segment;
 
  iommu->node = -1;
 
diff --git a/include/linux/intel-iommu.h b/include/linux/intel-iommu.h
index 2c4bed5..0a2da51 100644
--- a/include/linux/intel-iommu.h
+++ b/include/linux/intel-iommu.h
@@ -319,6 +319,7 @@ struct intel_iommu {
  int  agaw; /* agaw of this iommu */
  int  msagaw; /* max sagaw of this iommu */
  unsigned int  irq;
+ u16  segment;     /* PCI segment# */
  unsigned char  name[13];    /* Device Name */
 
 #ifdef CONFIG_INTEL_IOMMU
-- 
1.7.1